sqlmesh/core/renderer.py

```diff
@@ -251,6 +256,17 @@ def _expand(node: exp.Expression) -> exp.Expression:
 
         return t.cast(exp.Subqueryable, query)
 
+    def time_column_filter(self, start: TimeLike, end: TimeLike) -> exp.Between:
+        """Returns a between statement with the properly formatted time column."""
+        if not self._time_column:
+            raise RuntimeError(
+                f"Cannot produce time column filter because model does not have a time column."
+            )
+
+        return exp.column(self._time_column.column)[
+            self._time_converter(start) : self._time_converter(end)  # type: ignore
```

Comment:
Please don't tell me this is the "between" DSL

Reply:
Please note that in Python the end index is exclusive so it doesn't map well to BETWEEN. Other issues with it aside.

Reply:
Can't this just be:
```
exp.column(self._time_column.column).between(...)
```

Reply:
will fix this once george's stuff is in